winter/schema/type_hint.py

```diff
@@ -0,0 +1,188 @@
+
+        if self.nullable:
+            data['x-nullable'] = True
+
+        if self.enum is not None:
+            data['enum'] = self.enum
+
+        if self.properties:
+            data['properties'] = {key: value.as_dict() for key, value in self.properties.items()}
+
+        return data
+
+
+# noinspection PyUnusedLocal
+@register(bool)
+def bool_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_BOOLEAN)
+
+
+# noinspection PyUnusedLocal
+@register(NoneType)
+def none_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(TYPE_NONE)
+
+
+# noinspection PyUnusedLocal
+@register(int)
+def int_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_INTEGER)
+
+
+# noinspection PyUnusedLocal
+@register(str)
+def str_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING)
+
+
+# noinspection PyUnusedLocal
+@register(float)
+def float_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_NUMBER)
+
+
+# noinspection PyUnusedLocal
+@register(dict)
+def dict_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_OBJECT)
+
+
+# noinspection PyUnusedLocal
+@register(decimal.Decimal)
+def decimal_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(TYPE_DECIMAL, openapi.FORMAT_DECIMAL)
+
+
+# noinspection PyUnusedLocal
+@register(uuid.UUID)
+def uuid_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_UUID)
+
+
+# noinspection PyUnusedLocal
+@register(datetime.datetime)
+def datetime_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_DATETIME)
+
+
+# noinspection PyUnusedLocal
+@register(datetime.date)
+def date_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_DATE)
+
+
+# noinspection PyUnusedLocal
+@register(list, tuple, collections.Iterable)
+def iterable_resolver(hint_class) -> YASGTypeInfo:
+    args = getattr(hint_class, '__args__', None)
+    child_class = args[0] if args else str
+    child_type_info = get_basic_type_info_from_hint(child_class)
+    return YASGTypeInfo(openapi.TYPE_ARRAY, child=child_type_info)
+
+
+@register(enum.IntEnum)
+def intenum_resolver(enum_class: typing.Type[enum.IntEnum]) -> YASGTypeInfo:
+    return enum_resolver(enum_class)
+
+
+@register(enum.Enum)
+def enum_resolver(enum_class: typing.Type[enum.Enum]) -> YASGTypeInfo:
+    enum_values = [entry.value for entry in enum_class]
+    # Try to infer type based on enum values
+    enum_value_types = {type(v) for v in enum_values}
+
+    if len(enum_value_types) == 1:
+        type_info = get_basic_type_info_from_hint(enum_value_types.pop())
+    else:
+        type_info = YASGTypeInfo(openapi.TYPE_STRING)
+    type_info.enum = enum_values
+    return type_info
+
+
+@register(UnionType, checker=is_optional)
+def union_resolver(hint_class) -> YASGTypeInfo:
```

Comment:
Rename `union_resolver` to `optional_resolver`

Reply:
ok